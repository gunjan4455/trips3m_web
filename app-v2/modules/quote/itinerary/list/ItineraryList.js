import React from 'react';
import PropTypes from 'prop-types';
import { Element } from 'react-scroll';
import { Link } from 'react-router-dom';
import styled from '@emotion/styled';

import { DownArrow, ItineraryIcon } from 'helpers/Icon/Icon';
import { PDM_CATEGORY_ITINERARY } from 'helpers/pdmHelpers';
import { itineraryPath } from 'helpers/urlHelpers';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import { ACTIVITY } from "constants/activityConstant";
import QuoteCardHeader from "modules/quote/shared/QuoteCardHeader";

const GItineraryArrowSpan = styled.span({
  transform: 'rotateZ(-90deg)'
});

const ItineraryList = ({
                         tripId,
                         quoteId,
                         itineraryInfo,
                         trackSegment,
                         setPdmCategory,
                         showHidePdmView
                       }) => {
  const trackSegmentEvent = (event, cta, object) => {
    trackSegment({
      event,
      section: 'itinerary',
      object: object || '',
      cta
    });
  };

  const handleChatClick = () => {
    setPdmCategory(PDM_CATEGORY_ITINERARY);
    showHidePdmView(true);
    trackSegmentEvent(ORGANISM_CLICKED, 'Edit Package', 'Main');
  };

  const renderListItems = (item, index) => {
    const itineraryUrl = itineraryPath(tripId, quoteId);
    const day = index + 1;

    return <Link to={`${itineraryUrl}?day=${day}`} key={index}>
      <div className='flex alignCenter spaceBetween p15 bt at_itinerarylist'>
        <div className='flexFull'>
          <div className='flex alignCenter'>
            <h4 className='f16 fw9 pfc3 m0 mr8'>{`Day ${day}`}</h4>
            {item.eventTime ? (
              <p className='f12 pfc4 m0 at_eventTime'>{item.eventTime}</p>
            ) : null}
          </div>
          <p className='f14 pfc3 mt5'>{item.title}</p>
        </div>
        <GItineraryArrowSpan className='input-24 block p5 flex alignCenter justifyCenter'>
          <DownArrow />
        </GItineraryArrowSpan>
      </div>
    </Link>;
  };

  return (
    <Element name='itinerary-sec'>
      <div className="pfc3 f14 sbcw mt8">
        <QuoteCardHeader
          heading='Itinerary'
          chatButtonName='Edit Package'
          handleChatClick={handleChatClick}
          icon={<ItineraryIcon />}
        />
        <div>
          {itineraryInfo.map((item, index) => (
            Object.keys(item).map((section) => {
              if (item[section].sectionType === ACTIVITY.itinerary) {
                return renderListItems(item[section], index);
              }
            })
          ))}
        </div>
      </div>
    </Element>
  );
};

ItineraryList.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  itineraryInfo: PropTypes.array,
  trackSegment: PropTypes.func,
  showHidePdmView: PropTypes.func,
  setPdmCategory: PropTypes.func
};
ItineraryList.defaultProps = {
  itineraryInfo: [],
  tracksegment: () => {},
  setPdmCategory: () => {},
  showHidePdmView: () => {}
};
export default ItineraryList;
