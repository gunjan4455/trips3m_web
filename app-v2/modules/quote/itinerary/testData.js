export const itineraryInfo = [
  {
    "0": {
      order: '1',
      title: 'Private Transfer from Airport to Hotel',
      description:
        'A representative will meet and take you to you to your hotel.',
      eventTime: '15:35 - India Standard Time - 2 hours',
      sectionType: 'cab',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ]
    },
    "1": {
      order: '4',
      title: 'Arrival at Dubai Airport.',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Check in to hotel.',
      eventTime: 'Monday, 24 August, 2019',
      sectionType: 'itinerary',
      imagesUrl: []
    },
    "2": {
      order: '2',
      title: 'hotel details',
      description: 'hotel detail',
      eventTime: 'hotel timing',
      sectionType: 'hotel',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ],
      otherDetails: {
        name: 'hotel',
        rating: 5,
        email: '',
        phone: '44445454',
        web_url: '',
        category: '',
        hotelDesc: 'hotel description',
        tripAdvisorId: '',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "3": {
      order: '3',
      title: 'Night Safari',
      description:
        "At night, you will be taken for a memorable Night Safari experience. Being the first wildlife night park, it shows you nocturnal animalsâ€™ behavior in their natural habitat. This open-air zoo is known to house nearly 130 species of animals. After an exceptional experience, return to your hotel and stay overnight. Tip: Don't feed animals at the Night Safari",
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    },

    "4": {
      order: '4',
      title: 'Arrival',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Chk in to hotel.',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'flight',
      otherDetails: {
        name: 'prashant',
        image_url:
          'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        address: 'abc',
        rating: 4.3,
        phone: '33434',
        web_url: '',
        airport_des: 'airport description',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "5": {
      order: '3',
      title: 'Back to Hotel',
      description: 'going back to hotel after sight seeing',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    }
  }
];
