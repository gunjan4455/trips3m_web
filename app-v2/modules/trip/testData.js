export const tripsData = {
  loaded: true,
  loading: false,
  allIds: [
    '4809460',
    '4982568',
    '4982590',
    '4982978'
  ],
  byId: {
    '4809460': {
      id: 4809460,
      tripDays: 5,
      toLoc: 'Sri Lanka',
      firstDestinationName: 'Sri Lanka',
      startDate: '2019-07-04',
      quotesCount: 3,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '4 Star'
      },
      budget: '38000..41000',
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Cancelled',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919480946000'
    },
    '4982568': {
      id: 4982568,
      tripDays: 9,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-11',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '3 Star, 2 Star'
      },
      budget: null,
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Dormant',
      child: 0,
      tripStage: 2,
      currency: null,
      phoneNo: '+919498256800'
    },
    '4982590': {
      id: 4982590,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Dormant',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498259000'
    },
    '4982978': {
      id: 4982978,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 3,
      bookingStatusLabel: 'Active',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Active',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498297800'
    }
  }
};