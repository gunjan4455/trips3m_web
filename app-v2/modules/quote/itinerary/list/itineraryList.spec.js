import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ItineraryList from './ItineraryList';
import { itineraryInfo } from '../testData';

configure({ adapter: new Adapter() });

const mockTrackSegment = jest.fn();

const DEFAULT_PROPS = {
  tripId: 111111,
  quoteId: 222222,
  itineraryInfo: itineraryInfo,
  tracksegment: mockTrackSegment
};

const shallowWrappperItineraryComponent = props => {
  return shallow(<ItineraryList {...props} />);
};
describe('Itinerary: Render Correctly ', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperItineraryComponent(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show itinerary of one day', () => {
    const wrapper = shallowWrappperItineraryComponent(props);
    expect(wrapper.find('.at_itinerarylist')).toHaveLength(1);
  });

  it('should not render itinerary when sectionType not present in props', () => {
    props.itineraryInfo[0][1].sectionType = 'cab';
    const wrapper = shallowWrappperItineraryComponent(props);
    expect(wrapper.find('.at_itinerarylist')).toHaveLength(0);
  });
});
