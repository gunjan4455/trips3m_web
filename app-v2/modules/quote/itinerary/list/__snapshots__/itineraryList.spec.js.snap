// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Itinerary: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItineraryList
    itineraryInfo={
      Array [
        Object {
          "0": Object {
            "description": "A representative will meet and take you to you to your hotel.",
            "eventTime": "15:35 - India Standard Time - 2 hours",
            "imagesUrl": Array [
              "https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg",
              "https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg",
            ],
            "order": "1",
            "sectionType": "cab",
            "title": "Private Transfer from Airport to Hotel",
          },
          "1": Object {
            "description": "Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Check in to hotel.",
            "eventTime": "Monday, 24 August, 2019",
            "imagesUrl": Array [],
            "order": "4",
            "sectionType": "itinerary",
            "title": "Arrival at Dubai Airport.",
          },
          "2": Object {
            "description": "hotel detail",
            "eventTime": "hotel timing",
            "imagesUrl": Array [
              "https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg",
            ],
            "order": "2",
            "otherDetails": Object {
              "category": "",
              "email": "",
              "hotelDesc": "hotel description",
              "lat": "28.489743",
              "long": "77.074602",
              "name": "hotel",
              "phone": "44445454",
              "rating": 5,
              "tripAdvisorId": "",
              "web_url": "",
            },
            "sectionType": "hotel",
            "title": "hotel details",
          },
          "3": Object {
            "description": "At night, you will be taken for a memorable Night Safari experience. Being the first wildlife night park, it shows you nocturnal animalsâ€™ behavior in their natural habitat. This open-air zoo is known to house nearly 130 species of animals. After an exceptional experience, return to your hotel and stay overnight. Tip: Don't feed animals at the Night Safari",
            "eventTime": "15:35 - India Standard Time",
            "imagesUrl": Array [],
            "order": "3",
            "sectionType": "attraction",
            "title": "Night Safari",
          },
          "4": Object {
            "description": "Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Chk in to hotel.",
            "eventTime": "15:35 - India Standard Time",
            "order": "4",
            "otherDetails": Object {
              "address": "abc",
              "airport_des": "airport description",
              "image_url": "https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg",
              "lat": "28.489743",
              "long": "77.074602",
              "name": "prashant",
              "phone": "33434",
              "rating": 4.3,
              "web_url": "",
            },
            "sectionType": "flight",
            "title": "Arrival",
          },
          "5": Object {
            "description": "going back to hotel after sight seeing",
            "eventTime": "15:35 - India Standard Time",
            "imagesUrl": Array [],
            "order": "3",
            "sectionType": "attraction",
            "title": "Back to Hotel",
          },
        },
      ]
    }
    quoteId={222222}
    setPdmCategory={[Function]}
    showHidePdmView={[Function]}
    tracksegment={[MockFunction]}
    tripId={111111}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="pfc3 f14 sbcw mt8"
      >
        <QuoteCardHeader
          category=""
          chatButtonName="Edit Package"
          handleChatClick={[Function]}
          heading="Itinerary"
          icon={<IconSvg />}
        />
        <div>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/requested_trips/111111/quotes/222222/itinerary?day=1"
          >
            <div
              className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
            >
              <div
                className="flexFull"
              >
                <div
                  className="flex alignCenter"
                >
                  <h4
                    className="f16 fw9 pfc3 m0 mr8"
                  >
                    Day 1
                  </h4>
                  <p
                    className="f12 pfc4 m0 at_eventTime"
                  >
                    Monday, 24 August, 2019
                  </p>
                </div>
                <p
                  className="f14 pfc3 mt5"
                >
                  Arrival at Dubai Airport.
                </p>
              </div>
              <ForwardRef(render)
                className="input-24 block p5 flex alignCenter justifyCenter"
              >
                <IconSvg />
              </ForwardRef(render)>
            </div>
          </Link>
        </div>
      </div>,
      "name": "itinerary-sec",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <QuoteCardHeader
            category=""
            chatButtonName="Edit Package"
            handleChatClick={[Function]}
            heading="Itinerary"
            icon={<IconSvg />}
          />,
          <div>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/requested_trips/111111/quotes/222222/itinerary?day=1"
            >
              <div
                className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
              >
                <div
                  className="flexFull"
                >
                  <div
                    className="flex alignCenter"
                  >
                    <h4
                      className="f16 fw9 pfc3 m0 mr8"
                    >
                      Day 1
                    </h4>
                    <p
                      className="f12 pfc4 m0 at_eventTime"
                    >
                      Monday, 24 August, 2019
                    </p>
                  </div>
                  <p
                    className="f14 pfc3 mt5"
                  >
                    Arrival at Dubai Airport.
                  </p>
                </div>
                <ForwardRef(render)
                  className="input-24 block p5 flex alignCenter justifyCenter"
                >
                  <IconSvg />
                </ForwardRef(render)>
              </div>
            </Link>
          </div>,
        ],
        "className": "pfc3 f14 sbcw mt8",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "category": "",
            "chatButtonName": "Edit Package",
            "handleChatClick": [Function],
            "heading": "Itinerary",
            "icon": <IconSvg />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                undefined,
                <Link
                  onlyActiveOnIndex={false}
                  style={Object {}}
                  to="/requested_trips/111111/quotes/222222/itinerary?day=1"
                >
                  <div
                    className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
                  >
                    <div
                      className="flexFull"
                    >
                      <div
                        className="flex alignCenter"
                      >
                        <h4
                          className="f16 fw9 pfc3 m0 mr8"
                        >
                          Day 1
                        </h4>
                        <p
                          className="f12 pfc4 m0 at_eventTime"
                        >
                          Monday, 24 August, 2019
                        </p>
                      </div>
                      <p
                        className="f14 pfc3 mt5"
                      >
                        Arrival at Dubai Airport.
                      </p>
                    </div>
                    <ForwardRef(render)
                      className="input-24 block p5 flex alignCenter justifyCenter"
                    >
                      <IconSvg />
                    </ForwardRef(render)>
                  </div>
                </Link>,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
                >
                  <div
                    className="flexFull"
                  >
                    <div
                      className="flex alignCenter"
                    >
                      <h4
                        className="f16 fw9 pfc3 m0 mr8"
                      >
                        Day 1
                      </h4>
                      <p
                        className="f12 pfc4 m0 at_eventTime"
                      >
                        Monday, 24 August, 2019
                      </p>
                    </div>
                    <p
                      className="f14 pfc3 mt5"
                    >
                      Arrival at Dubai Airport.
                    </p>
                  </div>
                  <ForwardRef(render)
                    className="input-24 block p5 flex alignCenter justifyCenter"
                  >
                    <IconSvg />
                  </ForwardRef(render)>
                </div>,
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/requested_trips/111111/quotes/222222/itinerary?day=1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="flexFull"
                    >
                      <div
                        className="flex alignCenter"
                      >
                        <h4
                          className="f16 fw9 pfc3 m0 mr8"
                        >
                          Day 1
                        </h4>
                        <p
                          className="f12 pfc4 m0 at_eventTime"
                        >
                          Monday, 24 August, 2019
                        </p>
                      </div>
                      <p
                        className="f14 pfc3 mt5"
                      >
                        Arrival at Dubai Airport.
                      </p>
                    </div>,
                    <ForwardRef(render)
                      className="input-24 block p5 flex alignCenter justifyCenter"
                    >
                      <IconSvg />
                    </ForwardRef(render)>,
                  ],
                  "className": "flex alignCenter spaceBetween p15 bt at_itinerarylist",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          className="flex alignCenter"
                        >
                          <h4
                            className="f16 fw9 pfc3 m0 mr8"
                          >
                            Day 1
                          </h4>
                          <p
                            className="f12 pfc4 m0 at_eventTime"
                          >
                            Monday, 24 August, 2019
                          </p>
                        </div>,
                        <p
                          className="f14 pfc3 mt5"
                        >
                          Arrival at Dubai Airport.
                        </p>,
                      ],
                      "className": "flexFull",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <h4
                              className="f16 fw9 pfc3 m0 mr8"
                            >
                              Day 1
                            </h4>,
                            <p
                              className="f12 pfc4 m0 at_eventTime"
                            >
                              Monday, 24 August, 2019
                            </p>,
                          ],
                          "className": "flex alignCenter",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Day 1",
                              "className": "f16 fw9 pfc3 m0 mr8",
                            },
                            "ref": null,
                            "rendered": "Day 1",
                            "type": "h4",
                          },
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": "Monday, 24 August, 2019",
                              "className": "f12 pfc4 m0 at_eventTime",
                            },
                            "ref": null,
                            "rendered": "Monday, 24 August, 2019",
                            "type": "p",
                          },
                        ],
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Arrival at Dubai Airport.",
                          "className": "f14 pfc3 mt5",
                        },
                        "ref": null,
                        "rendered": "Arrival at Dubai Airport.",
                        "type": "p",
                      },
                    ],
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <IconSvg />,
                      "className": "input-24 block p5 flex alignCenter justifyCenter",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "__emotion_base": "span",
                      "__emotion_forwardProp": undefined,
                      "__emotion_real": [Circular],
                      "__emotion_styles": Array [
                        Object {
                          "transform": "rotateZ(-90deg)",
                        },
                      ],
                      "defaultProps": undefined,
                      "displayName": "Styled(span)",
                      "render": [Function],
                      "withComponent": [Function],
                    },
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            undefined,
            undefined,
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="pfc3 f14 sbcw mt8"
        >
          <QuoteCardHeader
            category=""
            chatButtonName="Edit Package"
            handleChatClick={[Function]}
            heading="Itinerary"
            icon={<IconSvg />}
          />
          <div>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/requested_trips/111111/quotes/222222/itinerary?day=1"
            >
              <div
                className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
              >
                <div
                  className="flexFull"
                >
                  <div
                    className="flex alignCenter"
                  >
                    <h4
                      className="f16 fw9 pfc3 m0 mr8"
                    >
                      Day 1
                    </h4>
                    <p
                      className="f12 pfc4 m0 at_eventTime"
                    >
                      Monday, 24 August, 2019
                    </p>
                  </div>
                  <p
                    className="f14 pfc3 mt5"
                  >
                    Arrival at Dubai Airport.
                  </p>
                </div>
                <ForwardRef(render)
                  className="input-24 block p5 flex alignCenter justifyCenter"
                >
                  <IconSvg />
                </ForwardRef(render)>
              </div>
            </Link>
          </div>
        </div>,
        "name": "itinerary-sec",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <QuoteCardHeader
              category=""
              chatButtonName="Edit Package"
              handleChatClick={[Function]}
              heading="Itinerary"
              icon={<IconSvg />}
            />,
            <div>
              <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/requested_trips/111111/quotes/222222/itinerary?day=1"
              >
                <div
                  className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
                >
                  <div
                    className="flexFull"
                  >
                    <div
                      className="flex alignCenter"
                    >
                      <h4
                        className="f16 fw9 pfc3 m0 mr8"
                      >
                        Day 1
                      </h4>
                      <p
                        className="f12 pfc4 m0 at_eventTime"
                      >
                        Monday, 24 August, 2019
                      </p>
                    </div>
                    <p
                      className="f14 pfc3 mt5"
                    >
                      Arrival at Dubai Airport.
                    </p>
                  </div>
                  <ForwardRef(render)
                    className="input-24 block p5 flex alignCenter justifyCenter"
                  >
                    <IconSvg />
                  </ForwardRef(render)>
                </div>
              </Link>
            </div>,
          ],
          "className": "pfc3 f14 sbcw mt8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "category": "",
              "chatButtonName": "Edit Package",
              "handleChatClick": [Function],
              "heading": "Itinerary",
              "icon": <IconSvg />,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                Array [
                  undefined,
                  <Link
                    onlyActiveOnIndex={false}
                    style={Object {}}
                    to="/requested_trips/111111/quotes/222222/itinerary?day=1"
                  >
                    <div
                      className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
                    >
                      <div
                        className="flexFull"
                      >
                        <div
                          className="flex alignCenter"
                        >
                          <h4
                            className="f16 fw9 pfc3 m0 mr8"
                          >
                            Day 1
                          </h4>
                          <p
                            className="f12 pfc4 m0 at_eventTime"
                          >
                            Monday, 24 August, 2019
                          </p>
                        </div>
                        <p
                          className="f14 pfc3 mt5"
                        >
                          Arrival at Dubai Airport.
                        </p>
                      </div>
                      <ForwardRef(render)
                        className="input-24 block p5 flex alignCenter justifyCenter"
                      >
                        <IconSvg />
                      </ForwardRef(render)>
                    </div>
                  </Link>,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "class",
                "props": Object {
                  "children": <div
                    className="flex alignCenter spaceBetween p15 bt at_itinerarylist"
                  >
                    <div
                      className="flexFull"
                    >
                      <div
                        className="flex alignCenter"
                      >
                        <h4
                          className="f16 fw9 pfc3 m0 mr8"
                        >
                          Day 1
                        </h4>
                        <p
                          className="f12 pfc4 m0 at_eventTime"
                        >
                          Monday, 24 August, 2019
                        </p>
                      </div>
                      <p
                        className="f14 pfc3 mt5"
                      >
                        Arrival at Dubai Airport.
                      </p>
                    </div>
                    <ForwardRef(render)
                      className="input-24 block p5 flex alignCenter justifyCenter"
                    >
                      <IconSvg />
                    </ForwardRef(render)>
                  </div>,
                  "onlyActiveOnIndex": false,
                  "style": Object {},
                  "to": "/requested_trips/111111/quotes/222222/itinerary?day=1",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="flexFull"
                      >
                        <div
                          className="flex alignCenter"
                        >
                          <h4
                            className="f16 fw9 pfc3 m0 mr8"
                          >
                            Day 1
                          </h4>
                          <p
                            className="f12 pfc4 m0 at_eventTime"
                          >
                            Monday, 24 August, 2019
                          </p>
                        </div>
                        <p
                          className="f14 pfc3 mt5"
                        >
                          Arrival at Dubai Airport.
                        </p>
                      </div>,
                      <ForwardRef(render)
                        className="input-24 block p5 flex alignCenter justifyCenter"
                      >
                        <IconSvg />
                      </ForwardRef(render)>,
                    ],
                    "className": "flex alignCenter spaceBetween p15 bt at_itinerarylist",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <div
                            className="flex alignCenter"
                          >
                            <h4
                              className="f16 fw9 pfc3 m0 mr8"
                            >
                              Day 1
                            </h4>
                            <p
                              className="f12 pfc4 m0 at_eventTime"
                            >
                              Monday, 24 August, 2019
                            </p>
                          </div>,
                          <p
                            className="f14 pfc3 mt5"
                          >
                            Arrival at Dubai Airport.
                          </p>,
                        ],
                        "className": "flexFull",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              <h4
                                className="f16 fw9 pfc3 m0 mr8"
                              >
                                Day 1
                              </h4>,
                              <p
                                className="f12 pfc4 m0 at_eventTime"
                              >
                                Monday, 24 August, 2019
                              </p>,
                            ],
                            "className": "flex alignCenter",
                          },
                          "ref": null,
                          "rendered": Array [
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Day 1",
                                "className": "f16 fw9 pfc3 m0 mr8",
                              },
                              "ref": null,
                              "rendered": "Day 1",
                              "type": "h4",
                            },
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Monday, 24 August, 2019",
                                "className": "f12 pfc4 m0 at_eventTime",
                              },
                              "ref": null,
                              "rendered": "Monday, 24 August, 2019",
                              "type": "p",
                            },
                          ],
                          "type": "div",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Arrival at Dubai Airport.",
                            "className": "f14 pfc3 mt5",
                          },
                          "ref": null,
                          "rendered": "Arrival at Dubai Airport.",
                          "type": "p",
                        },
                      ],
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": <IconSvg />,
                        "className": "input-24 block p5 flex alignCenter justifyCenter",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "__emotion_base": "span",
                        "__emotion_forwardProp": undefined,
                        "__emotion_real": [Circular],
                        "__emotion_styles": Array [
                          Object {
                            "transform": "rotateZ(-90deg)",
                          },
                        ],
                        "defaultProps": undefined,
                        "displayName": "Styled(span)",
                        "render": [Function],
                        "withComponent": [Function],
                      },
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
